components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/EventPayloadCryptographyUtils.java

```diff
@@ -153,6 +156,9 @@ private static PublicKey getPublicKey(String tenantDomain) throws ParseException
         } catch (IOException e) {
             throw new IdentityEventException("Unable to fetch event encryption public key for tenant " +
                     tenantDomain, e);
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+            throw new IdentityEventException(
+                    "Unable to generate RSA public key from the retrieved key due to invalid algorithm.", e);
```

Comment:
Since the Algorithm is fixed, This exception won't get hit. So didn't add much details to the log.

Reply:
Lets catch the invalidkeyexception separately and add an appropriate message. The key endpoint could return a bad key and lead to errors here too.